### PR TITLE
update to River v0.2.0 for raw strings

### PR DESCRIPTION
**Pull request by erikbaranowski**
#### PR Checklist

- [x] CHANGELOG.md updated
- [x] Documentation added
- [ ] Tests updated
- [ ] Config converters updated

### CODE REVIEW

File: docs/sources/flow/config-language/expressions/types_and_values.md [hunk 1]
```diff
@@ -88,6 +88,30 @@ The supported escape sequences are as follows:
 | `\uNNNN` | A Unicode character from the basic multilingual plane (NNNN is four hexadecimal digits) |
 | `\UNNNNNNNN` | A Unicode character from supplementary planes (NNNNNNNN is eight hexadecimal digits) |
 
+## Raw Strings
```
Comment (rfratto) [inline docs/sources/flow/config-language/expressions/types_and_values.md:91]: Maybe this should explain that raw strings do not support any escape sequences somewhere?

Comment (rfratto) [inline docs/sources/flow/config-language/expressions/types_and_values.md:91]: Use sentence case:

```suggestion
## Raw strings
```

File: go.sum [hunk 2]
```diff
@@ -1082,6 +1082,8 @@ github.com/grafana/regexp v0.0.0-20221123153739-15dc172cd2db h1:7aN5cccjIqCLTzed
 github.com/grafana/regexp v0.0.0-20221123153739-15dc172cd2db/go.mod h1:M5qHK+eWfAv8VR/265dIuEpL3fNfeC21tXXp9itM24A=
 github.com/grafana/river v0.1.2-0.20231003183959-75f893ffa7df h1:jgguNFI2lCCOD49W1WS+TYHhN5KL5aTXTbn2N14tQVs=
 github.com/grafana/river v0.1.2-0.20231003183959-75f893ffa7df/go.mod h1:icSidCSHYXJUYy6TjGAi/D+X7FsP7Gc7cxvBUIwYMmY=
+github.com/grafana/river v0.2.0 h1:VgBe/+DdYviJ1cYGUCcA8Og0FhSTjN7jdHCe/+OIFV0=
```
Comment (rfratto) [inline go.sum:1085]: This should also probably be a changelog entry in the CHANGELOG; not the dependency update itself but a feature entry talking about support for raw strings in River.

**Review by rfratto — APPROVED** [review]

File: go.sum (hunk 2 above)
Comment (erikbaranowski) [reply go.sum:1085]: sorry, it helps when you push all your commits! let me know if that covers it

Comment (rfratto) [reply go.sum:1085]: Yep that's fine 👍

**Review by clayton-cornell — APPROVED** [review]
looks OK to me

File: docs/sources/flow/config-language/expressions/types_and_values.md [hunk 3]
```diff
@@ -88,6 +88,31 @@ The supported escape sequences are as follows:
 | `\uNNNN` | A Unicode character from the basic multilingual plane (NNNN is four hexadecimal digits) |
 | `\UNNNNNNNN` | A Unicode character from supplementary planes (NNNNNNNN is eight hexadecimal digits) |
 
+## Raw strings
+
+Raw strings are represented by sequences of Unicode characters surrounded by backticks ``` `` ```.
```
Comment (tpaschalis) [inline docs/sources/flow/config-language/expressions/types_and_values.md:93]: What did we intend to show here with the multiple backticks? I don't think it renders as we wanted it to.

File: docs/sources/flow/config-language/expressions/types_and_values.md [hunk 4]
```diff
@@ -88,6 +88,31 @@ The supported escape sequences are as follows:
...
+## Raw strings
+
+Raw strings are represented by sequences of Unicode characters surrounded by backticks ``` `` ```. 
+Raw strings do not support any escape sequences:
+
+```river
+`Hello, "world"!`
+```
+
+Within the backticks, any character may appear except a backtick.
```
Comment (tpaschalis) [inline docs/sources/flow/config-language/expressions/types_and_values.md:100]: Do we want to mention that if someone _does_ want to insert a backtick, they can do so using string concatenation?

**Review by tpaschalis — APPROVED** [review]

File: docs/sources/flow/config-language/expressions/types_and_values.md (hunk 3 above)
Comment (erikbaranowski) [reply docs/sources/flow/config-language/expressions/types_and_values.md:93]: This mirrors the description of `Strings` above.

image

image

Comment (tpaschalis) [reply docs/sources/flow/config-language/expressions/types_and_values.md:93]: Ah, ok, didn't realize Hugo would render it correctly, thanks!

**Pull request merged**